tests/tests_integration/test_api/test_hosted_extractors/test_sources.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+import pytest
+
+from cognite.client import CogniteClient
+from cognite.client.data_classes.hosted_extractors import (
+    EventHubSource,
+    EventHubSourceUpdate,
+    EventHubSourceWrite,
+    SourceList,
+)
+from cognite.client.exceptions import CogniteAPIError
+from cognite.client.utils._text import random_string
+
+
+@pytest.fixture(scope="session")
+def one_event_hub_source(cognite_client: CogniteClient) -> SourceList:
+    my_hub = EventHubSourceWrite(
+        external_id=f"myNewHub-{random_string(10)}",
+        host="myHost",
+        key_name="myKeyName",
+        key_value="myKey",
+        event_hub_name="myEventHub",
+    )
+    retrieved = cognite_client.hosted_extractors.sources.retrieve(my_hub.external_id, ignore_unknown_ids=True)
+    if retrieved:
+        return retrieved
```

Comment:
This will probably never trigger since you use `random_string(10)`. What about 
```py
f"{platform.system()}-{platform.python_version()"
```

Reply:
Maybe something I am missing here, but why would I need that if I have a random string?

Reply:
Point is you will never retrieve it when using a random string, hence make each test runner have its own

Reply:
Thanks finally got it now.